frontend/lib/CMakeLists.txt

```diff
@@ -45,10 +59,40 @@ add_subdirectory(types)
 add_subdirectory(uast)
 add_subdirectory(util)
 
-add_library(libdyno SHARED $<TARGET_OBJECTS:libdyno-obj>)
+add_library(libdyno SHARED $<TARGET_OBJECTS:libdyno-obj> $<TARGET_OBJECTS:git-sha-obj>)
+
 target_include_directories(libdyno PUBLIC
                            ${CHPL_MAIN_INCLUDE_DIR}
                            ${CHPL_INCLUDE_DIR})
+
 target_link_libraries(libdyno PUBLIC ${CHPL_LLVM_LINK_ARGS})
 # TODO: Get printchplenv output proper so that we don't need to SHELL here
 target_compile_options(libdyno PUBLIC SHELL:$<$<COMPILE_LANGUAGE:CXX>:${CHPL_LLVM_COMP_ARGS}>)
+
+# TODO: the ideal way to cleanup would be to add ${CMAKE_CURRENT_SOURCE_DIR}/util/git-version.cpp
+# to the list of BYPRODUCTS, but that creates a strong cycle between libdyno and git-sha-obj
+# which is not allowed when the libraries are not static. The workaround so far is to set
+# the ADDITIONAL_CLEAN_FILES property of libdyno to add git-version.cpp, but that property
+# wasn't added until 3.15 and our reqs list 3.13.4
+# generate the git-version.cpp and compile it to produce the .o before linking libdyno
+# If .git doesn't exist, we are probably in a release tarball or Dockerfile
```

Comment:
From this comment my understanding is that if we are building in a release tarball or Docker image, we expect that `git-version.cpp` has been previously generated and is present. However I don't see how this would be true for our release tarballs or Docker images currently -- i.e. how when making a release tarball are we getting `git-version.cpp` in there without manual intervention? Similarly for the Docker build, `make` only occurs within the Docker build context which does not have `.git` so where is `git-version.cpp` coming from?

Reply:
> how when making a release tarball are we getting git-version.cpp in there without manual intervention?

Interesting, we've probably gotten away with it because we aren't skipping `doc` generation, so `make docs` is called, which writes `git-version.cpp` when linking to the compiler library. However, if we did skip the docs by setting `CHPL_GEN_RELEASE_SKIP_DOCS`, then we might see the file missing, as we would with the `Dockerfile` if the `cmake` configure step had never been ran before doing the `docker build` command.

I think we can solve it in `gen_release` by forcing the utility script that builds the `git-version.cpp` file to run by calling it explicitly. For the `Dockerfile`, it seems like we'd either have to copy the `.git` folder, OR wrap the `docker build` command in a build script that first calls the `write-git-sha` script as well.

Reply:
Ok, I think that solution makes sense for `gen_release`. For Docker I don't think there's any way for us to get around including `.git` without violating the self-contained nature of the `Dockerfile` and image, so I would go ahead and get rid of the `.dockerignore`. This will increase the size of the image, which I don't think is a big deal, but we should keep it in mind and potentially look into something like a multi-stage Docker build to reduce size if it becomes a concern in the future.